### DISCUSSION TITLE
New error on latest release build

### DISCUSSION THREAD

**Discussion by gwww** [wezterm/wezterm#2481] [General]
I just updated to the latest release build (Sept 3 build) and now get this on startup. I don't see anything in the release notes that point to what is happening. Pointers?
```
Error converting lua value returned by script /Users/glenn/.config/wezterm/
wezterm.lua to Config struct: error converting Lua table to Config (Error
processing keys.action.args (types: Config, Key, SpawnCommand) Cannot convert
`Object` to `Vec`)
```
The config line where it's happening is here: https://github.com/gwww/dotfiles/blob/main/.config/wezterm/keys.lua#L10

Thanks!

**Comment by wez** [comment]
You're not using `wezterm.action` to wrap that up. I don't believe that we've changed anything around this since the release in June (https://github.com/wez/wezterm/issues/1150 specifically).

I'd recommend changing that line to:

```lua
  return wezterm.action.SpawnCommandInNewTab{cwd=wezterm.config_dir, args=args}
```

**Comment by gwww** [comment]
I have narrowed it down. `wezterm.action` did not help (but, its a good change to make regardless). The bit that is not working is:
```
local args = { os.getenv("SHELL"), "-li", "-c", "exec $VISUAL " .. files }
```
Specifically, the `os.getenv("SHELL")`, I believe is returning nothing. I tested it by replacing the `os.getenv...` with `"/opt/homebrew/bin/zsh"` and that works. The only thing I can figure is this commit removes SHELL environment variable: https://github.com/wez/wezterm/commit/ad9490ee8ff2c39d7df2f9996055555a9a538b59 (which fixes this https://github.com/wez/wezterm/issues/2378 issue).

**Reply by wez** [reply]
Ah, interesting. 7158c435d5a56bfa1502dd0af7470a9cca868c81 should resolve that in a robust way!

**Comment by gwww** [comment]
Thank you! I tested it and it works!
